### DISCUSSION TITLE
Channel worker, error in redis asyncio

### DISCUSSION THREAD

**Discussion by YonatanNemtsov** [django/channels#1967] [Help!]
Hi, I have a consumer and "worker" consumer, connected in a redis channel layer.

When I run the server and worker I sometimes get the following error in the server:

File ".../lib/python3.10/site-packages/redis/asyncio/client.py", line 536, in parse_response
    retval = self.response_callbacks[command_name](response, **options)
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'

as far as I understand, parse_response, defined [here](https://github.com/redis/redis-py/blob/master/redis/asyncio/client.py), parses a response from the redis server and returns it. So I want to ask, Is this an redis-py bug, or channels bug, or am I doing something wrong?

Here are the consumer and "worker" consumer:
```
class GameConsumer(AsyncJsonWebsocketConsumer):
    async def connect(self):
        await self.accept()
        await self.channel_layer.send(
            'test',
            {
                'type':'f',
                'message':'Hello',
                'channel_name':self.channel_name,
            }

        )
        return
```

```
# worker
class Test(AsyncConsumer):
    async def f(self, message):
        print(message)
```

Thanks

**Comment by carltongibson** [comment]
> as far as I understand, parse_response, defined [here](https://github.com/redis/redis-py/blob/master/redis/asyncio/client.py), parses a response from the redis server and returns it.

It's hard to see from what you've given — but looks like it's between redis and Python, so redis-py at a guess. 

Can you reduce it and open an issue over there to see what they say? 
(Full details of the locals at the frame would be helpful.)

**Reply by YonatanNemtsov** [reply]
Yes, thank you.
